Sources/Basics/Archiver+Tar.swift

```diff
@@ -0,0 +1,147 @@
+
+/// An `Archiver` that handles Tar archives using the command-line `tar` tool.
+public final class TarArchiver: Archiver {
+    public var supportedExtensions: Set<String> { ["tar", "tar.gz"] }
+
+    /// The file-system implementation used for various file-system operations and checks.
+    private let fileSystem: FileSystem
+
+    /// Helper for cancelling in-flight requests
+    private let cancellator: Cancellator
+
+    /// The underlying command
+    private let tarCommand: String
+
+    /// Creates a `TarArchiver`.
+    ///
+    /// - Parameters:
+    ///   - fileSystem: The file system to used by the `TarArchiver`.
+    ///   - cancellator: Cancellation handler
+    public init(fileSystem: FileSystem, cancellator: Cancellator? = .none) {
+        self.fileSystem = fileSystem
+        self.cancellator = cancellator ?? Cancellator(observabilityScope: .none)
+
+        #if os(Windows)
+        self.tarCommand = "tar.exe"
+        #else
+        self.tarCommand = "tar"
+        #endif
+    }
+
+    public func extract(
+        from archivePath: AbsolutePath,
+        to destinationPath: AbsolutePath,
+        completion: @escaping (Result<Void, Error>) -> Void
+    ) {
+        do {
+            guard self.fileSystem.exists(archivePath) else {
+                throw FileSystemError(.noEntry, archivePath)
+            }
+
+            guard self.fileSystem.isDirectory(destinationPath) else {
+                throw FileSystemError(.notDirectory, destinationPath)
+            }
+
+            let process = TSCBasic.Process(
+                arguments: [self.tarCommand, "xzf", archivePath.pathString, "-C", destinationPath.pathString]
+            )
+
+            guard let registrationKey = self.cancellator.register(process) else {
+                throw StringError("cancellation")
+            }
+
+            DispatchQueue.sharedConcurrent.async {
+                defer { self.cancellator.deregister(registrationKey) }
+                completion(.init(catching: {
+                    try process.launch()
+                    let processResult = try process.waitUntilExit()
+                    guard processResult.exitStatus == .terminated(code: 0) else {
+                        throw try StringError(processResult.utf8stderrOutput())
+                    }
+                }))
+            }
+        } catch {
+            return completion(.failure(error))
+        }
+    }
+
+    public func compress(
+        directory: AbsolutePath,
+        to destinationPath: AbsolutePath,
+        completion: @escaping (Result<Void, Error>) -> Void
+    ) {
+        do {
+            guard self.fileSystem.isDirectory(directory) else {
+                throw FileSystemError(.notDirectory, directory)
+            }
+
+            let process = TSCBasic.Process(
+                arguments: [self.tarCommand, "acf", destinationPath.pathString, directory.basename],
+                workingDirectory: directory.parentDirectory
+            )
```

Comment:
true on all platforms?

Reply:
Yes, as clarified above by Saleem. I also checked macOS manpages that these arguments have the same meaning. AFAIU `tar.exe` on Windows follows the convention.